libbeat/processors/actions/alterFieldProcessor.go

```diff
@@ -0,0 +1,175 @@
+package actions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	conf "github.com/elastic/elastic-agent-libs/config"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+)
+
+// alterFieldFunc defines how fields must be processed
+type alterFieldFunc func(field string) string
+
+type alterFieldProcessor struct {
+	Fields        []string
+	IgnoreMissing bool
+	FailOnError   bool
+
+	processorName string
+	alterFunc     alterFieldFunc
+}
+
+// NewAlterFieldProcessor is an umbrella method for processing events based on provided fields. Such as converting event keys to uppercase/lowercase
+func NewAlterFieldProcessor(c *conf.C, processorName string, alterFunc alterFieldFunc) (processors.Processor, error) {
+	config := struct {
+		Fields        []string `config:"fields"`
+		IgnoreMissing bool     `config:"ignore_missing"`
+		FailOnError   bool     `config:"fail_on_error"`
+		FullPath      bool     `config:"full_path"`
```

Comment:
If the implementation is not going to be a part of this PR, the configuration should contain this option either.